app/src/main/java/com/verygoodsecurity/demoshow/ui/activity/ImageActivity.kt

```diff
@@ -0,0 +1,108 @@
+import org.json.JSONObject
+
+class ImageActivity : AppCompatActivity(R.layout.activity_image),
+    VgsCollectResponseListener, VGSOnResponseListener {
+
+    private val show: VGSShow by lazy {
+        VGSShow.Builder(this, MainActivity.TENANT_ID).build()
+    }
+
+    private val collect: VGSCollect by lazy {
+        VGSCollect(this, MainActivity.TENANT_ID, MainActivity.ENVIRONMENT)
+    }
+
+    private val vgsImageView: VGSImageView by lazy { findViewById(R.id.vgsImageView) }
+    private val flProgress: FrameLayout by lazy { findViewById(R.id.flProgress) }
+    private val tvFileAlias: TextView by lazy { findViewById(R.id.tvFileAlias) }
+    private val tvFileName: TextView by lazy { findViewById(R.id.tvFileName) }
+    private val mbAttachFile: MaterialButton by lazy { findViewById(R.id.mbAttachFile) }
+    private val mbSubmit: MaterialButton by lazy { findViewById(R.id.mbSubmit) }
+    private val mbReveal: MaterialButton by lazy { findViewById(R.id.mbReveal) }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setupCollect()
+        setupShow()
+    }
+
+    @Suppress("OVERRIDE_DEPRECATION", "DEPRECATION")
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        collect.onActivityResult(requestCode, resultCode, data)
+        updateAttachedFileName()
+    }
+
+    override fun onResponse(response: CollectResponse) {
+        Log.d(PDFActivity::class.java.simpleName, response.toString())
+        flProgress.visibility = View.GONE
+        (response as? CollectSuccessResponse)?.rawResponse?.let {
+            try {
+                JSONObject(it).getJSONObject("json").getString("image").run {
+                    tvFileAlias.text = this
+                }
+            } catch (e: Exception) {
+                Log.d(ImageActivity::class.simpleName, e.message ?: "Invalid json or data not exist!")
+            }
+        }
+    }
+
+    override fun onResponse(response: ShowResponse) {
+        Log.d(PDFActivity::class.java.simpleName, response.toString())
+        flProgress.visibility = View.GONE
+    }
+
+    private fun setupCollect() {
+        collect.addOnResponseListeners(this)
+
+        mbAttachFile.setOnClickListener {
+            collect.getFileProvider().attachFile(this, "image")
+        }
+        mbSubmit.setOnClickListener {
+            flProgress.visibility = View.VISIBLE
+            collect.asyncSubmit("/post", HTTPMethod.POST)
+        }
+    }
+
+    private fun setupShow() {
+        show.subscribe(vgsImageView)
+        show.addOnResponseListener(this)
+
+        mbReveal.setOnClickListener {
+            flProgress.visibility = View.VISIBLE
+            show.requestAsync(
+                VGSRequest.Builder("post", VGSHttpMethod.POST)
+                    .body(mapOf("revealed_image" to (tvFileAlias.text ?: "tok_sandbox_9tXw3pQdLUq25Lm4bVpMkC")))
```

Comment:
I think this hardcoded token will never work `tok_sandbox_9tXw3pQdLUq25Lm4bVpMkC` maybe it is a better idea to show a message or something like that so the user knows it is failing?

Reply:
@DonaldRG Those hardcoded token was added for testing, thanks.